test/Npgsql.Tests/Types/PostgisTest.cs

```diff
@@ -0,0 +1,235 @@
+        {
+            using (var cmd = Conn.CreateCommand())
+            {
+                var a = _geoms[geomIdx];
+                cmd.CommandText = "Select " + a.SQL;
+                var p = cmd.ExecuteScalar();
+                Assert.IsTrue(p.Equals(a.Geom));
+            }
+        }
+
+        [Test]
+        [TestCase("point")]
+        [TestCase("line")]
+        [TestCase("poly")]
+        [TestCase("mpoint")]
+        [TestCase("mline")]
+        [TestCase("mpoly")]
+        [TestCase("coll")]
+        [TestCase("nestedcoll")]
+        public void PostgisTestWrite(string geomIdx)
+        {
+            using (var cmd = Conn.CreateCommand())
+            {                
+                cmd.Parameters.AddWithValue("p1", NpgsqlTypes.NpgsqlDbType.Geometry,_geoms[geomIdx].Geom);
+                _geoms[geomIdx].Geom.SRID = 0;
+                cmd.CommandText = "Select st_asewkb(:p1) = st_asewkb(" + _geoms[geomIdx].SQL + ")";
+                Assert.IsTrue((bool)cmd.ExecuteScalar());
+            }
+        }
+
+        [Test]
+        [TestCase("point")]
+        [TestCase("line")]
+        [TestCase("poly")]
+        [TestCase("mpoint")]
+        [TestCase("mline")]
+        [TestCase("mpoly")]
+        [TestCase("coll")]
+        [TestCase("nestedcoll")]
+        public void PostgisTestWriteSrid(string geomIdx)
+        {
+            using (var cmd = Conn.CreateCommand())
+            {
+                cmd.Parameters.AddWithValue("p1", NpgsqlTypes.NpgsqlDbType.Geometry, _geoms[geomIdx].Geom);
+                _geoms[geomIdx].Geom.SRID = 3942;
+                cmd.CommandText = "Select st_asewkb(:p1) = st_asewkb(st_setsrid("+ _geoms[geomIdx].SQL + ",3942))";
+                var p = (bool)cmd.ExecuteScalar();
+                Assert.IsTrue(p);
+            }
+        }
+
+        [Test]
+        [TestCase("point")]
+        [TestCase("line")]
+        [TestCase("poly")]
+        [TestCase("mpoint")]
+        [TestCase("mline")]
+        [TestCase("mpoly")]
+        [TestCase("coll")]
+        [TestCase("nestedcoll")]
+        public void PostgisTestReadSrid(string geomIdx)
+        {
+            using (var cmd = Conn.CreateCommand())
+            {
+                var a = _geoms[geomIdx];
+                cmd.CommandText = "Select st_setsrid(" + a.SQL + ",3942)";
+                var p = cmd.ExecuteScalar();
+                Assert.IsTrue(p.Equals(a.Geom));
+                Assert.IsTrue((p as IGeometry).SRID == 3942);
+            }
+        }
+        
+        protected override void SetUp()
+        {
+            base.SetUp();
+            using (var cmd = Conn.CreateCommand())
+            {
+                cmd.CommandText = "SELECT postgis_version();";
+                try
+                {
+                    cmd.ExecuteNonQuery();
+                }
+                catch (NpgsqlException)
+                {
+                    Assert.Ignore("Skipping tests : postgis extension not found.");
```

Comment:
Please use `TestUtil.IgnoreExceptOnBuildServer`, this makes sure the tests fail and aren't ignored on the build server.